XenModel/Actions/Updates/DownloadAndUnzipXenServerPatchAction.cs

```diff
@@ -165,11 +167,11 @@ private void ExtractFile()
 
                     while (iterator.HasNext())
                     {
-                        string currentExtension = Path.GetExtension(iterator.CurrentFileName());
+                        string currentExtension = Path.GetExtension(iterator.CurrentFileName()).Replace(".","");
 
                         if (Array.Exists(updateFileExtensions, item => item == currentExtension))
                         {
-                            string path = address != null ? Path.Combine(Path.GetDirectoryName(zippedFileName), iterator.CurrentFileName())
+                            string path = !UnzipOnly ? Path.Combine(Path.GetDirectoryName(zippedFileName), iterator.CurrentFileName())

```

Comment:
I know that I suggested the UnzipOnly variable, but seeing now that it is always used as `!UnzipOnly`, maybe having a variable with the opposite meaning would be more efficient? 
(I don't have a strong opinion about it, so if we cannot come up with a better variable name, then it's fine as it is.)